src/cmdstan/write_datetime.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef CMDSTAN_WRITE_DATETIME_HPP
+#define CMDSTAN_WRITE_DATETIME_HPP
+
+#include <stan/callbacks/writer.hpp>
+#include <stan/version.hpp>
+#include <chrono>
+#include <iomanip>
+#include <string>
+
+namespace cmdstan {
+
+void write_datetime(stan::callbacks::writer& writer) {
+  const std::time_t current_datetime
+      = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+  std::tm* curr_tm = std::localtime(&current_datetime);
+  std::stringstream current_datetime_msg;
+  current_datetime_msg << "start_datetime = " << std::setfill('0')
+                       << (1900 + curr_tm->tm_year) << "-" << std::setw(2)
+                       << (curr_tm->tm_mon + 1) << "-" << std::setw(2)
+                       << curr_tm->tm_mday << " " << std::setw(2)
+                       << curr_tm->tm_hour << ":" << std::setw(2)
+                       << curr_tm->tm_min << ":" << std::setw(2)
+                       << curr_tm->tm_sec;
+  writer(current_datetime_msg.str());
```

Comment:
This is not pretty and can be replaced by 
```cpp
const std::time_t current_datetime = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
std::stringstream current_datetime_msg;
current_datetime_msg << std::put_time(std::localtime(&current_datetime), "%F %T");
writer("start_datetime = " + current_datetime_msg.str());
```
once we up the minimum required g++ to 5+. g++ 4.9 does not have `std::put_time` even though its part of C++11...

I also tried using `std::strftime` and char arrays but that produced a few garbage characters.